tcp_test.go

```diff
@@ -40,30 +41,125 @@ func (t testTun) WritePacket(p *tuntap.Packet) error {
 	return t.write_error
 }
 
-type testTCPData struct {
+type testNode struct {
 	in          chan types.Packet
 	out         chan types.Packet
 	write_error error
 }
 
-func (d testTCPData) SendPacket(p types.Packet) error {
+func (n testNode) SendPacket(p types.Packet) error {
+	n.in <- p
+	return n.write_error
+}
+func (n testNode) Packets() <-chan types.Packet {
+	return n.out
+}
+func (n testNode) GetNodeAddress() types.NodeAddress {
+	return "source"
+}
+
+func readTunPacket() tuntap.Packet {
```

Comment:
That wouldn't make sense. This is supposed to be an IP packet coming from a tun device.